src/nvim/api/buffer.c

```diff
@@ -162,6 +163,12 @@ Boolean nvim_buf_attach(uint64_t channel_id, Buffer buffer, Boolean send_buffer,
     return false;
   }
 
+  // load buffer first if it's not loaded
+  if (!buf_ensure_loaded(buf)) {
```

Comment:
IMO this should not be done. "attach" expresses interests in callbacks about the buffer. It should be possible to pre-register this interest in an unloaded buffer without any change of state (and then have events start arriving once it is loaded). I think this is already how it works as well.

Reply:
agreed, and therefore removed this one